packages/run-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -672,19 +716,34 @@ const managerBehavior = {
 
 const collateralBehavior = {
   /** @param {MethodContext} context */
-  makeVaultInvitation: ({ state: { zcf }, facets: { self } }) =>
-    zcf.makeInvitation(self.makeVaultKit, 'MakeVault'),
+  makeVaultInvitation: ({ state, facets: { self } }) => {
+    const { zcf } = provideEphemera(state.collateralBrand);
+    assert(zcf);
+    return zcf.makeInvitation(self.makeVaultKit, 'MakeVault');
+  },
   /** @param {MethodContext} context */
-  getSubscriber: ({ state }) => state.assetSubscriber,
+  getSubscriber: ({ state }) => {
+    const { assetSubscriber } = provideEphemera(state.collateralBrand);
+    assert(assetSubscriber);
+    return assetSubscriber;
+  },
   /** @param {MethodContext} context */
-  getMetrics: ({ state }) => state.metricsSubscription,
+  getMetrics: ({ state }) => {
+    const { metricsSubscription } = provideEphemera(state.collateralBrand);
+    assert(metricsSubscription);
+    return metricsSubscription;
+  },
   /** @param {MethodContext} context */
   getCompoundedInterest: ({ state }) => state.compoundedInterest,
 };
 
 const selfBehavior = {
   /** @param {MethodContext} context */
-  getGovernedParams: ({ state }) => state.factoryPowers.getGovernedParams(),
+  getGovernedParams: ({ state }) => {
+    const { factoryPowers } = provideEphemera(state.collateralBrand);
+    assert(factoryPowers);
```

Comment:
I think `assert()` without a message is pretty hard to track down. Once there's a message, it becomes much easier, even when the message is duplicated within the source file. If I'm right about that, it would be worth changing all of these to e.g.
```suggestion
    assert(factoryPowers, 'factoryPowers');
```

Reply:
I agree that would be better. I intend to make it so the assertion isn't necessary so I'm opting to invest the effort there instead.